comtypes/test/test_portabledevice.py

```diff
@@ -0,0 +1,129 @@
+            clsctx=comtypes.CLSCTX_INPROC_SERVER,
+            interface=port_api.IPortableDeviceManager,
+        )
+        p_id_cnt = pointer(c_ulong())
+        mng.GetDevices(POINTER(c_wchar_p)(), p_id_cnt)
+        if p_id_cnt.contents.value == 0:
+            self.skipTest("There is no portable device in the environment.")
+        dev_ids = (c_wchar_p * p_id_cnt.contents.value)()
+        mng.GetDevices(cast(dev_ids, POINTER(c_wchar_p)), p_id_cnt)
+        self.device = comtypes.client.CreateObject(
+            port_api.PortableDevice().IPersist_GetClassID(),
+            clsctx=comtypes.CLSCTX_INPROC_SERVER,
+            interface=port_api.IPortableDevice,
+        )
+        self.device.Open(list(dev_ids)[0], info)
+
+    def test_EnumObjects(self):
+        WPD_OBJECT_PROPERTIES_V1 = GUID("{EF6B490D-5CD8-437A-AFFC-DA8B60EE4A3C}")
+        WPD_OBJECT_NAME = PropertyKey(WPD_OBJECT_PROPERTIES_V1, 4)
+        WPD_OBJECT_CONTENT_TYPE = PropertyKey(WPD_OBJECT_PROPERTIES_V1, 7)
+        WPD_OBJECT_SIZE = PropertyKey(WPD_OBJECT_PROPERTIES_V1, 11)
+        WPD_OBJECT_ORIGINAL_FILE_NAME = PropertyKey(WPD_OBJECT_PROPERTIES_V1, 12)
+        WPD_OBJECT_PARENT_ID = PropertyKey(WPD_OBJECT_PROPERTIES_V1, 3)
+        folderType = GUID("{27E2E392-A111-48E0-AB0C-E17705A05F85}")
+        functionalType = GUID("{99ED0160-17FF-4C44-9D98-1D7A6F941921}")
+
+        content = self.device.Content()
+        properties = content.Properties()
+        key_collection = comtypes.client.CreateObject(
+            port_types.PortableDeviceKeyCollection,
+            clsctx=comtypes.CLSCTX_INPROC_SERVER,
+            interface=port_api.IPortableDeviceKeyCollection,
+        )
+        key_collection.Add(WPD_OBJECT_NAME)
+        key_collection.Add(WPD_OBJECT_CONTENT_TYPE)
+
+        def findDir(curObjectId: str, level: int) -> Optional[str]:
+            """Searches the directory given by the first parameter for a sub-directory at a given level,
+            and returns its object ID if it exists"""
+            if level < 0:
+                return None
+            values = properties.GetValues(curObjectId, key_collection)
+            contenttype = values.GetGuidValue(WPD_OBJECT_CONTENT_TYPE)
+            is_folder = contenttype in [folderType, functionalType]
+            # print(f"{values.GetStringValue(WPD_OBJECT_NAME)} ({'folder' if is_folder else 'file'})")
+            if is_folder:
+                if level == 0:
+                    return curObjectId
+                # traverse into the children
+                enumobj = content.EnumObjects(c_ulong(0), curObjectId, None)
+                for x in enumobj:
+                    objId = findDir(x, level - 1)
+                    if objId:
+                        return objId
+                return None  # not in this part of the tree
+            else:
+                return None
+
+        # find a directory 2 levels deep, because you can't write on the top ones
+        parentId = findDir("DEVICE", 2)
```

Comment:
The variable name should be the PEP-8ish `parent_id`.

The same applies to other variable names.